xla/service/gpu/transforms/collective_permute_cycle_decomposer_test.cc

```diff
@@ -188,46 +218,86 @@ TEST_F(CollectivePermuteCycleDecomposerTest, ForwardCycleWithMatmul) {
+    // CHECK-SAME{LITERAL}: {{0,3}}, frontend_attributes={_xla_send_recv_validation={{0,7}}}, metadata={op_name="op1/op2/add" source_file="foo/bar/mysource.py" source_line=35}
+    
+    // CHECK:   %[[cp2:.+]] = u32[8,8] collective-permute(%{{.+}}), channel_id=2, source_target_pairs=
+    // CHECK-SAME{LITERAL}: {{1,0},{2,1},{3,2}}, frontend_attributes={_xla_send_recv_validation={{1,8},{2,9},{3,10}}}, metadata={op_name="op1/op2/add" source_file="foo/bar/mysource.py" source_line=35}
+    
+    // CHECK:   ROOT %{{.+}} = u32[8,8] select(%[[compare]], %[[cp1]], %[[cp2]])
+    // CHECK: }
+  )"));
+}
+
+TEST_F(CollectivePermuteCycleDecomposerTest, BackwardCycleNoChannel) {
+  // For backward cycle, this checks:
+  // 1. Split collectives do not have a channel-id
+  // 2. Split collectives are combined based on the value of replica-id.
+  const absl::string_view kModuleStr = R"(
+    HloModule test
+    ENTRY test_computation {
+      p = u32[8,8] parameter(0)
+      ROOT start = u32[8,8] collective-permute(p),
+        source_target_pairs={{0,3},{1,0},{2,1},{3,2}},
+        frontend_attributes={_xla_send_recv_validation="{{0,7},{1,8},{2,9},{3,10}}"}
+    })";
+
+  TF_ASSERT_OK_AND_ASSIGN(std::unique_ptr<HloModule> module,
+                          ParseAndReturnVerifiedModule((kModuleStr)));
   CollectivePermuteCycleDecomposer decomposer(/*threshold_in_bytes=*/0);
   TF_ASSERT_OK_AND_ASSIGN(bool changed, decomposer.Run(module.get()));
   EXPECT_TRUE(changed);
-  auto check_metadata = [](const HloInstruction* inst) {
-    EXPECT_EQ(inst->metadata().op_name(), "op1/op2/add");
-    EXPECT_EQ(inst->metadata().source_file(), "foo/bar/mysource.py");
-    EXPECT_EQ(inst->metadata().source_line(), 35);
-  };
-
-  HloCollectivePermuteInstruction* cp1 =
-      DynCast<HloCollectivePermuteInstruction>(
-          FindInstruction(module.get(), "collective-permute"));
-  HloCollectivePermuteInstruction* cp2 =
-      DynCast<HloCollectivePermuteInstruction>(
-          FindInstruction(module.get(), "collective-permute.1"));
-  EXPECT_NE(cp1, nullptr);
-  EXPECT_NE(cp2, nullptr);
-  EXPECT_EQ(cp1->operand(0), cp2->operand(0));
-  EXPECT_GT(cp2->channel_id().value(), cp1->channel_id().value());
-  EXPECT_THAT(cp1->ToString(), HasSubstr("source_target_pairs={{0,3}}"));
-  EXPECT_THAT(cp1->ToString(), HasSubstr("_xla_send_recv_validation={{0,7}}"));
-  EXPECT_THAT(cp2->ToString(),
-              HasSubstr("source_target_pairs={{1,0},{2,1},{3,2}}"));
-  EXPECT_THAT(cp2->ToString(),
-              HasSubstr("_xla_send_recv_validation={{1,8},{2,9},{3,10}}"));
-  check_metadata(cp1);
-  check_metadata(cp2);
+  HloPrintOptions options;
+  options.set_print_operand_shape(false);
+  options.set_include_layout_in_shapes(false);
+  TF_CHECK_OK(VerifyHloModule(module.get(), false, true));
+  EXPECT_TRUE(*RunFileCheck(module->ToString(options), R"(
+    // CHECK: ENTRY %test_computation (p: u32[8,8]) -> u32[8,8] {
```

Comment:
With indentation, I meant that `ENTRY %test_computation` and `%[[replica_id:` are aligned

Reply:
It is 
```
ENTRY %test_computation ...
  %[[replica_id:...
```
Because of the fact that `CHECK-DAG` is longer than `CHECK`, it is just a coincidence that `%test_computation` aligns with `%[[replica_id:.*]]`. Do you want me to remove the three (3) spaces before replica id, or add more spaces before the statements within the computation?

Reply:
I thought to add spaces before `ENTRY %t` to make it align with the closing `// CHECK-DAG: }` at the end and to do the same for the checks in between. Basically to use spaces to account for the differnet length of `CHECK`, `CHECK-DAG` and friends. 

Like xla/service/gpu/fusions/concatenate_mlir_test.cc#109

Reply:
Done. Thanks!